### PR TITLE
Update godocs after refactor

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

File: pkg/controller/example_test.go [hunk 1]
```diff
@@ -30,7 +30,21 @@ import (
 
 var mrg manager.Manager
 
-// This example creates a new controller named "pod-controller" with a no-op reconcile function and registers
+// This example creates a new Controller named "pod-controller" with a no-op reconcile function.  The
+// manager.Manager mrg will be used to Start the Controller, and will provide it a shared Cache and Client.
```
Comment (droot) [inline pkg/controller/example_test.go]: nit: mrg --> mgr (looks like your IDE has picked `mrg` instead of `mgr`) ?

File: pkg/doc.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+Example 2: source.Channel reads Generic events from a channel fed by a WebHook called from a Slackbot.
+
+* source provides a stream of events for EventHandlers to handle.
+
+* source may provide either events from Watches (e.g. object Create, Update, Delete) or Generic triggered
+from another source (e.g. WebHook callback).
+
+* Users SHOULD use the provided Source implementations instead of implementing their own for nearly all cases.
+
+EventHandler
```
Comment (droot) [inline pkg/doc.go]: We made changes to naming in `EventHandler` I think ?

File: pkg/doc.go [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+Event Multiplexing:
+
+* reconcile.Requests for the same name / Namespace are deduplicated when they are enqueued.  This allows
+for Controllers to gracefully handle event storms for a single object.  Multiplexing multiple event Sources to
+a single object type takes advantage of this.
+
+* Example: Pod events for a ReplicaSet are transformed to a ReplicaSet name / Namespace, so the ReplicaSet
+will be Reconciled only 1 time for multiple Pods.
+*/
```
Comment (droot) [inline pkg/doc.go]: This is a nice doc write up. +1

File: pkg/doc.go (hunk 2 above)
Comment (pwittrock) [reply pkg/doc.go]: The interface stil EventHandler

**Pull request merged**